sdks/python/apache_beam/typehints/decorators.py

```diff
@@ -309,38 +309,54 @@ def with_input_types(*positional_hints, **keyword_hints):
   be type-hinted in totality if even one parameter is type-hinted.
 
   Once fully decorated, if the arguments passed to the resulting function
-  violate the type-hint constraints defined, a TypeCheckError detailing the
-  error will be raised.
+  violate the type-hint constraints defined, a :class:`TypeCheckError`
+  detailing the error will be raised.
 
-  To be used as::
+  To be used as:
 
-    * @with_input_types(s=str)  # just @with_input_types(str) will work too.
-      def upper(s):
-        return s.upper()
+  .. testcode::
 
-  Or::
+    from apache_beam.typehints import with_input_types
 
-    * @with_input_types(ls=List[Tuple[int, int])
-      def increment(ls):
-        [(i + 1, j + 1) for (i,j) in ls]
+    @with_input_types(s=str)  # just @with_input_types(str) will work too.
+    def upper(s):
+      return s.upper()
+
+  Or:
+
+  .. testcode::
```

Comment:
Every code snippet should be tested. It's not testing any functionality in particular, it will just help us to make sure the code examples aren't outdated and still run/compile.